### PR TITLE
Implement Vertex finder with one kernel instead of 5

### PR DESCRIPTION
**Pull request by VinInn**
After realizing that the number of vertex to split is small I implemented the whole sequence
finder, fitter,split,fitter,ptsort as a single kernel of ONE block.
to make things simple only for the Density clustering.
Other options are still available using configurable (or conditional compilation).

The quadruplet workflow is a couple of percent (one...) faster on T4.
```
                    5.75%  564.21ms      5000  112.84us  4.8640us  741.69us  gpuVertexFinder::vertexFinderOneKernel(ZVertexSoA*, gpuVertexFinder::WorkSpace*, int, float, float, float)
      API calls:   25.46%  4.66845s    175000  26.676us  6.3150us  61.718ms  cudaLaunchKernel
                   18.76%  3.43955s     13678  251.47us  1.6050us  5.9472ms  cudaEventSynchronize

   965.6 ±   2.4 ev/s

------
                    2.81%  265.91ms      5000  53.181us  3.0080us  786.65us  gpuVertexFinder::clusterTracksByDensityKernel(ZVertexSoA*, gpuVertexFinder::WorkSpace*, int, float, float, float)
                    1.90%  180.09ms      5000  36.017us     992ns  131.74us  gpuVertexFinder::sortByPt2Kernel(ZVertexSoA*, gpuVertexFinder::WorkSpace*)
                    0.63%  59.975ms     10000  5.9970us  1.2800us  120.26us  gpuVertexFinder::fitVerticesKernel(ZVertexSoA*, gpuVertexFinder::WorkSpace*, float)
                    0.29%  27.200ms      5000  5.4400us  1.5680us  215.36us  gpuVertexFinder::splitVerticesKernel(ZVertexSoA*, gpuVertexFinder::WorkSpace*, float)
      API calls:   29.07%  5.72791s    195000  29.373us  6.0080us  68.871ms  cudaLaunchKernel
                   18.05%  3.55706s    211228  16.839us     586ns  68.982ms  cudaEventRecord

   952.6 ±   2.1 ev/s
```

### CODE REVIEW

**Comment by fwyzard** [comment]
:construction: Validation running at fu-c2a02-37-01:/data/user/fwyzard/patatrack/validation/run.y5YUoQAnSG ...

File: RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinderImpl.h [hunk 1]
```diff
@@ -77,17 +119,37 @@ namespace gpuVertexFinder {
 #endif
 
 #ifdef __CUDACC__
-    if (useDensity_) {
-      clusterTracksByDensity<<<1, 1024 - 256, 0, stream>>>(soa, ws_d.get(), minT, eps, errmax, chi2max);
+    if (oneKernel_) {
+      // implemented only for density clustesrs   
+#ifdef ONE_KERNEL
+      vertexFinderOneKernel<<<1, 1024 - 256, 0, stream>>>(soa, ws_d.get(), minT, eps, errmax, chi2max);
```
Comment (makortel) [inline RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinderImpl.h]: Is there a good motivation for the "one kernel" to be controlled both by `#ifdef` and by configuration parameter?

Comment (VinInn) [reply RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinderImpl.h]: ONE_KERNEL controls one vs three as before ONE was done using dynamic parallelism (some as three but in its own kernel) that at the moment does not even link...
I still hope at some point to have the possibility to test dynamic parallelism
(so ok, everything can be done at config level)
I can remove the three kernel option, is just historical development at this point...

Comment (makortel) [reply RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinderImpl.h]: Ok, I was just curious. I'm fine with leaving it in, but maybe add a comment explaining `ONE_KERNEL`?

Comment (VinInn) [reply RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinderImpl.h]: I will cleanup and comment in next iteration.

**Comment by fwyzard** [comment]
@VinInn , the throughput actually shows somewhat of a slowdown.

**Comment by VinInn** [comment]
we can revert to 5 kernels (enough to change one line)

**Comment by VinInn** [comment]
OOOPSSS a bug.  missing braces....
it was running the last 4 kernels twice..
(bug introduced with the latest commit to make it configurable at run time)

so run this morning in a new area using latest patatrack master
twice each version on T4 "Running 4 times over 5000 events with 1 jobs, each with 8 threads, 8 streams and 1 GPUs"
```
current:        980.7 ±   9.2 ev/s    986.0 ±   5.6 ev/s
oneKernel:      995.8 ±   9.5 ev/s    992.2 ±   0.6 ev/s
ThreeKernels:   990.3 ±   7.1 ev/s    994.7 ±   9.7 ev/s
FiveKernels:    985.7 ±   3.8 ev/s    989.5 ±   5.7 ev/s
```

**Comment by VinInn** [comment]
ok, need to rebase....

**Comment by fwyzard** [comment]
:construction: Validation running at fu-c2a02-35-01:/data/user/fwyzard/patatrack/validation/run.liAnaPM7Fb ...

**Pull request merged**